src/controller/content/content.service.ts

```diff
@@ -1853,4 +1912,212 @@ export class ContentService {
   }
 
 
+  async getAllQuestionTypes() {
+    try {
+      const allQuestionTypes = await db.select().from(zuvyQuestionTypes);
+      if (allQuestionTypes.length > 0) {
+        return {
+          status: 'success',
+          code: 200,
+          allQuestionTypes,
+        };
+      } else {
+        return [];
+      }
+    } catch (err) {
+      throw err;
+    }
+  }
+
+  async createQuestionType(questionType: CreateTypeDto) {
+    try {
+      const newQuestionType = await db.insert(zuvyQuestionTypes).values(questionType).returning();
+      if (newQuestionType.length > 0) {
+        return {
+          status: 'success',
+          code: 200,
+          newQuestionType,
+        };
+      } else {
+        return {
+          status: 'error',
+          code: 404,
+          message: 'Question Type is not created.Please try again.',
+        };
+      }
+    } catch (err) {
+      throw err;
+    }
+  }
+
+
+  async createFormForModule(form: formBatchDto) {
+    try {
+      const formQuestions = form.questions.map((f) => ({
+        question: f.question,
+        options: f.options,
+        typeId: f.typeId,
+      }));
+
+      for(let i=0; i<formQuestions.length;i++){
```

Comment:
Avoid using a for loop with an array. 
try this 
const allFieldsFilled = formQuestions.every(question => question.question !== null && question.options !== null && question.typeId !== null);
if (!allFieldsFilled) {   return {   status: "error",   code: 400,     message: "   One or more fields are empty. Please try again." };}